common/src/IO/MapFileSerializer.h

```diff
@@ -41,11 +41,18 @@ namespace TrenchBroom {
             using LineStack = std::vector<size_t>;
             LineStack m_startLineStack;
             size_t m_line;
-            FILE* m_stream;
+            std::ostream& m_stream;
+        protected:
+            /**
+             * Buffering strategy: to avoid the cost of writing every character (or every line)
+             * to the std::ostream, we write to this vector instead, and then transfer it in large
+             * blocks to m_stream (see flushBufferIfNeeded).
+             */
+            std::vector<char> m_buffer;
```

Comment:
`std::ostream` already does buffering. The buffer is flushed with `std::endline`, so if you use `\n` instead, it shouldn't flush so often and cache more. You can control the buffer size for the stream by calling `std::ostream::setbuf` I think.

Another approach might be to pass a `std::ostringstream`, have the whole map written to memory, and save that to a file in one go afterwards.

Reply:
I can double check those, but they're probably slower than what I have here. I tried 5 or 6 approaches. `std::ostream` is generally really slow with small writes, AFAIK just due to several layers of virtual function call indirection.

Reply:
It's not as bad as I expected, `fmt::format_to(std::ostreambuf_iterator<char>(m_stream), ...` is only around 5% slower (2422ms -> 2520ms) than my DIY buffering with the std::vector. So I'll go with ostreambuf_iterator since it trims out some code that we won't have to maintain later.